### PR TITLE
Unity test app. Send attachments in trackError.

### PR DESCRIPTION
**Pull request by olmisa**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [X] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

**Related issues**
[AB#71073](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/71073)

**Description**
Modify the Unity test apps to send attachments even when "handle errors" toggle is on. Also ensure that the attachments functionality remains enabled in this case.

### CODE REVIEW

File: Assets/Puppet/PuppetCrashes.cs [hunk 1]
```diff
@@ -146,6 +148,11 @@ public void LastCrashReport()
     }
 
     public static ErrorAttachmentLog[] GetErrorAttachmentstHandler(ErrorReport errorReport)
+    {
+        return GetErrorAttachmentst();
+    }
+
+    private static ErrorAttachmentLog[] GetErrorAttachmentst()
```
Comment (annakocheshkova) [inline Assets/Puppet/PuppetCrashes.cs]:
```suggestion
    private static ErrorAttachmentLog[] GetErrorAttachments()
```

File: Assets/Puppet/PuppetCrashes.cs [hunk 2]
```diff
@@ -146,6 +148,11 @@ public void LastCrashReport()
     }
 
     public static ErrorAttachmentLog[] GetErrorAttachmentstHandler(ErrorReport errorReport)
+    {
+        return GetErrorAttachmentst();
```
Comment (annakocheshkova) [inline Assets/Puppet/PuppetCrashes.cs]:
```suggestion
        return GetErrorAttachments();
```

**Review by annakocheshkova — CHANGES_REQUESTED** [review]

**Review by Jamminroot — DISMISSED** [review]

File: Assets/Puppet/PuppetCrashes.cs (hunk 1 above)
Comment (olmisa) [reply Assets/Puppet/PuppetCrashes.cs]: Fixed

File: Assets/Puppet/PuppetCrashes.cs (hunk 2 above)
Comment (olmisa) [reply Assets/Puppet/PuppetCrashes.cs]: Fixed

**Review by AnastasiaKubova — DISMISSED** [review]

**Comment by jaeklim** [comment]
Is it too much work to align with other test apps? Providing a way to select a binary and being able to type text attachment in the app? Its story point is 2 so I think this is doable.

File: Assets/Puppet/PuppetCrashes.cs [hunk 3]
```diff
@@ -146,6 +148,11 @@ public void LastCrashReport()
     }
 
     public static ErrorAttachmentLog[] GetErrorAttachmentstHandler(ErrorReport errorReport)
+    {
+        return GetErrorAttachments();
+    }
+
+    private static ErrorAttachmentLog[] GetErrorAttachments()
     {
         return new ErrorAttachmentLog[]
```
Comment (jaeklim) [inline Assets/Puppet/PuppetCrashes.cs]: Please check UI and return only valid attachments. This always returns two logs which are sent with unexpected data.

File: Assets/Puppet/PuppetCrashes.cs [hunk 4]
```diff
@@ -146,6 +148,11 @@ public void LastCrashReport()
...
         {
```
Comment (jaeklim) [inline Assets/Puppet/PuppetCrashes.cs]: Please read UI component before building attachments. It reads cached text and binary so test app doesn't send correct attachments when attachments are changed right before sending it.

**Review by jaeklim — CHANGES_REQUESTED** [review]

File: Assets/Puppet/PuppetCrashes.cs (hunk 4 above)
Comment (olmisa) [reply Assets/Puppet/PuppetCrashes.cs]: Fixed

File: Assets/Puppet/PuppetCrashes.cs [hunk 5]
```diff
@@ -146,11 +148,16 @@ public void LastCrashReport()
...
-            ErrorAttachmentLog.AttachmentWithText(PuppetAppCenter.TextAttachmentCached, "hello.txt"),
```
Comment (jaeklim) [inline Assets/Puppet/PuppetCrashes.cs:152]: Is `PuppetAppCenter.TextAttachmentCached` still used?

**Comment by guperrot** [comment]
Please merge `feature/handled-error-attachments` to this branch and retarget the PR.

**Comment by jaeklim** [comment]
Verified that attachment works for both Android and iOS but Select Binary button doesn't work on Android emulator.
Sorry for any of miscommunication for a functionality to select a file for binary attachment, what we have in other test apps is to open Photo and use a photo for binary attachment. I hate to say this one but please remove File picker and bring Photo or Gallery when the button is clicked.

**Review by jaeklim — APPROVED** [review]

**Pull request merged**